scs2-simulation/src/main/java/us/ihmc/scs2/simulation/physicsEngine/contactPointBased/ContactPointBasedRobotPhysics.java

```diff
@@ -160,4 +203,47 @@ public RobotPhysicsOutput getPhysicsOutput()
    {
       return physicsOutput;
    }
+
+   private void getJointTauFromControllers()
+   {
+      for (JointBasics joint : owner.getJointsToConsider())
+      {
+         JointStateBasics jointOutput = owner.getControllerOutput().getJointOutput(joint);
+
+         if (jointOutput.hasOutputFor(JointStateType.EFFORT))
+         {
+            int jointIndex = indexProvider.getJointDoFIndices(joint)[0];
+            jointOutput.getEffort(jointIndex, jointsTauController);
```

Comment:
I would grab the taus directly from the robot joints. It used to be with SCS1 that we would write the tau command directly as the robot's joint tau, I wouldn't be surprised if in some places in the codebase we doing the same with SCS2. So reading directly from the robot state is safer I'd say

Reply:
Removed the `jointsTauController` `DMatrix` variable completely, pulling straight from the joints as you say.